src/components/Search/Search.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC, useEffect, useRef, useState } from 'react';
+
+import css from './Seacrh.module.css';
+
+type ISearch = {
+  handleSearch: () => void;
+  setInputValue: (value: string) => void;
+};
+
+export const Search: FC<ISearch> = ({ setInputValue, handleSearch }) => {
+  const [input, setInput] = useState<string>(localStorage.getItem('inputValue') ?? '');
+  const searchRef = useRef(input);
+
+  useEffect(() => {
+    searchRef.current = input;
+  }, [input]);
+
+  useEffect(() => {
+    return () => {
+      localStorage.setItem('inputValue', searchRef.current);
+    };
+  }, []);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    setInput(event.target.value);
+  };
+
+  const handleKeyDown = async (event: React.KeyboardEvent<HTMLInputElement>): Promise<void> => {
+    if (event.code === 'Enter') {
+      await handleSearch();
```

Comment:
Refactor this redundant 'await' on a non-promise. Or fix the ISearch to return promise